### ISSUE TITLE
List of features for eventual backporting

### ISSUE THREAD

**Issue by SirCmpwn** [SirCmpwn/TrueCraft#27]
The current list of features that are planned for backporting. Suggest more in this thread.

* [ ] Fence gates
* [ ] Sprinting
* [ ] Beef
* [ ] Melons
* [ ] Stair corner rendering
* [ ] More stair and slab types
* [ ] Stone bricks
* [ ] Better lighting system
* [ ] New button recipe, wooden buttons
* [ ] Glass panes
* [ ] Abandoned mine shafts
* [ ] New bow mechanics
* [ ] Arrows that stick out of entities
* [ ] Prevent player-placed leaves from decaying
* [ ] 3D chest model and animation
* [ ] New inventory handling (painting, etc)
* [ ] Breeding
* [ ] Server browser

**Comment by NSDex** [comment]
Taller max world height.

**Comment by SirCmpwn** [comment]
> Taller max world height.

Agreed.

**Comment by UnknownShadow200** [comment]
Make biome generation server-side.

**Comment by SirCmpwn** [comment]
> Make biome generation server-side.

Yep.

**Comment by SirCmpwn** [comment]
Added `In-game list of online players`

**Comment by JimmyGibsJr** [comment]
All the redstone changes, like comparators, daylight sensors, slime blocks, etc.

**Comment by SirCmpwn** [comment]
>All the redstone changes, like comparators, daylight sensors, slime blocks, etc.

Nope.

**Comment by RockyTV** [comment]
Daylight sensors?

**Comment by SirCmpwn** [comment]
>Daylight sensors?

Nope.

**Comment by jleclanche** [comment]
Horses? (Not necessarily for riding)

**Comment by SirCmpwn** [comment]
>Horses? (Not necessarily for riding)

No.

**Comment by ConcernedHobbit** [comment]
Upside-down stairs, really useful for decorating

**Comment by SirCmpwn** [comment]
Agreed on the former, and the latter is a feature that existed in beta 1.7.3 and is going to be added anyway.

**Comment by zevipa** [comment]
can the 3D chest model be optional?

**Comment by SirCmpwn** [comment]
>can the 3D chest model be optional?

I don't see why not. We'll probably turn it off for a high performance mode (along with transparent leaves and such).

**Comment by SirCmpwn** [comment]
>please, please, please do not backport sprinting. sprinting broke combat completely, and monsters have not been a threat since sprinting was added. also make the beta 1.8 lighting optional. it is better in some ways, but worse in others. part of the charm of beta 1.7 was its lighting. same goes for bow mechanics. the old bows are completely different, and huge part of the old game. make the modern chest models optional as well. i like them, but i also like the old ones (this may mean having the 3d chests take up a full block, but they only recently changed that, so that shouldnt be a big deal).

**Comment by RockyTV** [comment]
Redstone block and lamp?
Pistons?
All the types of pressure plates?

**Comment by SirCmpwn** [comment]
>Redstone block and lamp?

Block no, lamp maybe.

> Pistons?

These are already present in beta 1.7.3.

> All the types of pressure plates?

I'll do stone and wooden and maybe an original one.

**Comment by jleclanche** [comment]
@SirCmpwn consider implementing redstone block as a decorative block only?

**Comment by SirCmpwn** [comment]
Probably not.

**Comment by robinkanters** [comment]
@jleclanche make it a mod when the API is ready, if you really want it :wink:

**Comment by phase** [comment]
> All the redstone changes, like *comparators*, daylight sensors, slime blocks, etc.

I'd really like to see just comparators implemented, they implement the ability to use hex in redstone. Adding everything else would be annoying, though.

**Comment by SirCmpwn** [comment]
>I'd really like to see just comparators implemented, they implement the ability to use hex in redstone. Adding everything else would be annoying, though.

Nope. The changes Mojang have made to redstone over the past few years are the worst of the changes they've made by far. The likelihood of them being backported is quite small.

**Comment by phase** [comment]
> the worst changes they've made by far.

Comparators are one of the only good changes they've made! It feels like a feature that should have been in the game for years.

**Comment by SirCmpwn** [comment]
Well, feel free to mod them into TrueCraft once we get modding support going.

**Comment by phase** [comment]
What language will the modding be in? C#?

**Comment by SirCmpwn** [comment]
Any .NET language. There are a lot of them.

**Comment by RockyTV** [comment]
Friendly mobs flee from you if you attack them? But with a better pathfinding?

**Comment by SirCmpwn** [comment]
The behavior of mob AI is not really a "feature" per-se, so long as, you know, skeletons fire arrows and creepers explode and what-not.

**Comment by Luigifan** [comment]
exploding arrows.

just please for the love of god never backport exp or hunger. two of the most obnoxious things added to beta 1.8 aside from the "swamp" biome. 

as for running, maybe we can make like a running shoes sort of item modification? idk how lenient you'd be on custom items.

**Comment by SirCmpwn** [comment]
>exploding arrows.

...was this ever a thing?

>just please for the love of god never backport exp or hunger. two of the most obnoxious things added to beta 1.8 aside from the "swamp" biome. 

Agreed.

>as for running, maybe we can make like a running shoes sort of item modification? idk how lenient you'd be on custom items.

I'm still inclined to add Mojang-style sprinting and give the ability to mobs as well to balance it out.

**Comment by jleclanche** [comment]
I like the swap biome :(

But yeah +1 on exp/hunger. What trash.

**Comment by SirCmpwn** [comment]
I don't mind the swap biome either, but I'm going to let you customize the biomes in use for your worlds.

**Comment by Luigifan** [comment]
>...was this ever a thing?
no just light banter

I think as long as it's balanced it'll work out fine. Also biome customisation works out great too. Awesome!

**Comment by Gbear605** [comment]
>>"exploding arrows."

>...was this ever a thing?

It's being added to 1.9, which is coming soon.

**Comment by SirCmpwn** [comment]
>EDIT: What about strongholds/mineshafts? I actually personally don't mind those too much and feel like they add a little bit of flavour to the world. Even if it's a custom generation option.

Mineshafts sure, strongholds maybe.

**Comment by SirCmpwn** [comment]
>It's being added to 1.9, which is coming soon.

Well, don't expect them in TrueCraft.

**Comment by Wulgaren** [comment]
Can you add creative mode?

**Comment by StianTheDark** [comment]
I will not support creative mode, if implemented. I'll probably make a mod to cut it out if it gets backported.

Also, please do not backport strongholds. They are boring, full of cheat-loot and don't feel natural at all. I wouldn't like mineshafts either, but I can't decide everything in this project.

**Comment by Wulgaren** [comment]
Why? It would be good just for building
16 sie 2015 22:02 "Stian Furu Øverbye" <notifications@github.com>
napisał(a):

**Comment by StianTheDark** [comment]
Gamemodes are a clutterf*ck, and if someone wants it then make a mod.

**Comment by jleclanche** [comment]
Some serious entitlement going on over here.

**Comment by RockyTV** [comment]
It's up to @SirCmpwn to decide to implement Creative mode, and strongholds. Personally, I think Creative should be added, but it should be a mod that is already included in TrueCraft's source, so you have to enable it if you want to use it. 

Same with strongholds.

**Comment by SirCmpwn** [comment]
Creative mode - yes
Strongholds - no
Mineshafts - yes

Justification for mineshafts - they encourage exploring underground. We can deal with balance issues separately if it's a problem.

**Comment by mrpimpunicorn** [comment]
Minecraft's first game mode _was_ creative, really; so it makes sense to back-port it. @SirCmpwn Did you manage to troubleshoot the previous issues with OpenGL, or are you fine with me resuming client development regardless?

**Comment by SirCmpwn** [comment]
I haven't spent any time on it, @mrpimpunicorn. I will get back to it soon, my focus at the moment is on other projects.

**Comment by Wulgaren** [comment]
Can you add trees species from new version and biomes such as jungle?

**Comment by SirCmpwn** [comment]
No.

**Comment by ShaunNoWay** [comment]
Could you add a method of legally obtaining ice blocks (without enchantments)? Or perhaps and new slippery block to be used soley for building purposes? I feel this was seriously missing from b1.7.3.

**Comment by SirCmpwn** [comment]
Probably, yes, but that falls under original features and is outside the scope of this issue.

**Comment by ShaunNoWay** [comment]
Oh, my bad. Should we discuss this on the Subreddit?

**Comment by SirCmpwn** [comment]
I suppose. I'd prefer to discuss it once we reach feature parity with beta 1.7.3, since it's certainly happening no earlier than that.

**Comment by ShaunNoWay** [comment]
May I ask why you want to backport beef? I think it was a better design choice back in 1.7.3 when each passive mob had their own unique drop: pigs for meat, cows for hide, sheep for wool and chickens for feathers. Having more than one mob drop food seems kinda superfluous, in my opinion.

**Comment by SirCmpwn** [comment]
Someone asked about beef, but I guess the comment was deleted? A greater variety of food is why I want beef, but also because it lines up with the vision by being both realistic and useful, as well as adding an interesting mechanic of cooking your food.

**Comment by jleclanche** [comment]
Are we still talking about minecraft?

**Comment by SirCmpwn** [comment]
Well, within a certain limit of "realism". One of my problems with modern MC is that it gets harder and harder to suspend your disbelief at the idea that you could wake up in the wilderness and manage to acquire all of the resources that modern MC lets you get. Beef doesn't hurt that.

**Comment by jleclanche** [comment]
I was making a joke. I'm hungry. :)

**Comment by ShaunNoWay** [comment]
I was the guy who posted about beef. I redacted it because it felt kinda bitchy to me at the time.
>May I ask why you want to backport beef? I think it was the better design choice back in 1.7.3 when each passive mob had their own unique drop: pigs for meat, cows for hide, sheep for wool and chickens for feathers. Having more than one mob drop food always seemed kinda superfluous, in my opinion.

Thanks for responding, SirCmpwn! I really appreciate it.

**Comment by ShaunNoWay** [comment]
Last question: What is "Better lighting system"?

**Comment by SirCmpwn** [comment]
Doing day/night cycle lighting on the fly instead of the update-the-world approach beta 1.7.3 takes. The b1.7.3 strategy is actually so stupid that I'm just not implementing it at all and we're going straight to the "better" way.

**Comment by ShaunNoWay** [comment]
Haha, okay. Thanks!

**Comment by bb010g** [comment]
> * [ ] 256 max world height

Any particular reason why it's 256?

**Comment by kiljacken** [comment]
Originally it was a size "optimization". A limit of 256, (or at the time 128) allowed a y-position (vertical axis) to be stored as a single byte on disk (and in the 128 days, a single signed bit in memory, due to java not having unsigned types).

**Comment by SirCmpwn** [comment]
Due to restrictions imposed by Minecraft. At some point we may be able to drop the vertical limit entirely.

**Comment by Alzarath** [comment]
I always thought all animals should drop meat in Minecraft. Even sheep. But I'd expect non-pig animals would drop less meat, as to keep pigs useful.

**Comment by SirCmpwn** [comment]
Propose that in #211

**Comment by kameko** [comment]
I hope some form of spriting or fast transport is implemented, regardless of what it is. I absolutely hated how slow the player walks and remember being overjoyed when sprinting was implemented, the days of going agonizingly slow from place to place were over.
Since hunger isn't a thing, maybe you can make running (be it a natural ability or through some kind of item) take health away instead? That would also discourage using running to get away from mobs, since it would weaken you and make you more vulnerable. Perhaps you could also implement running speed in the armor types (like leather goes 0.2x speed and diamond would be maybe 2.5x speed). Though I'm not sure how good an idea that is considering it takes like 20 minutes to get diamonds.

Also, what about hoppers? Those are post-b.1.7 right? They're a little more convenient than mile-long ice streams that send items into unloaded chunks (to later cause immense lag when you load them all in for collection).

**Comment by SirCmpwn** [comment]
>sprinting takes away health

No, but you will only be able to do it for a short amount of time

>running speed with armor

Nah

>hoopers

No
